optics-core/src/Optics.hs

```diff
@@ -109,15 +109,83 @@ module Optics
 
   -- * Optics utilities
 
-  -- ** Indexed optics
+  -- ** Re
 
-  -- | TODO
-  , module Optics.Indexed
+  -- | Some optics can be reversed with 're':
+  -- @'Iso' i s t a b@ into @'Iso' i b a t s@,
+  -- @'Getter' s t a b@ into @'Review' i b a t s@ etc.
+  -- Red arrows illustrate how 're' transforms optics:
+  --
+  -- <<reoptics.png Reversed Optics>>
+  --
+  -- 're' is mainly useful to invert 'Iso's:
+  --
+  -- >>> let _Identity = iso runIdentity Identity
+  -- >>> view1 (_1 % re _Identity) ('x', "yz")
+  -- Identity 'x'
+  --
+  -- Yet we can use a 'Lens' as a 'Review' too:
+  --
+  -- >>> review (re _1) ('x', "yz")
+  -- 'x'
+  --
+  -- /Note:/ there are no @from@ combinator.
 
-  -- ** Re
+  , module Optics.Re
 
-  -- | TODO
---  , module Optics.Re
+  -- ** Indexed optics
```

Comment:
I guess this section may need a bit of tweaking if we do #41.

Reply:
True. I'll plan to decide (and implement) #41 first.

Reply:
I made some corrections, but need to think properly how to approach indexed optics in the docs.